### ISSUE TITLE
Publisher doesn't appear to resume between restarts

### ISSUE THREAD

**Issue by stuartc** [slashdotdash/eventstore#8]
I can't quite seem to work out the best way to set up a Subscriber that receives events after a catch up.

I have GenServer with `EventStore.subscribe_to_all_streams("jobs_subscriber", self)`, and with a bit of logging can see previous events being notified as the server starts up.

However when I append events, the Publisher adds them to pending since it expects the first event to have been published:

```
[debug] attempting to create stream a1e1ac15-f799-4ce5-95a6-b5f33762a35f
[debug] created stream a1e1ac15-f799-4ce5-95a6-b5f33762a35f with id 39
[info] appended 1 events to stream id 39
[debug] Publisher.handle_cast/2 [:notify_events]
  expected_event_id: 1
  initial_event_id: 39
  last_event_id: 39
iex(9)> [debug] Publisher.handle_cast/2 [:notify_pending_events]
  last_published_event_id: 0
```

I can see in the AllStreamsSubscription module does a lot more than the example on the README. I mimic'd what I saw in the tests:

```elixir
    subscription =
      AllStreamsSubscription.new
      |> AllStreamsSubscription.subscribe("$all", to_string(__MODULE__), self)
      |> AllStreamsSubscription.catch_up
```

But when I append, the publisher still adds it to pending. How should I go about setting up a subscriber or the publisher?

**Comment by slashdotdash** [comment]
@stuartc You've found a bug there! The publisher attempts to publish events in order, but as you have seen it doesn't work after a restart. It is expecting an event with id = 1. On startup, I think it needs to go to the database and retrieve the latest event id so it knows what event to expect next. 

There is a chance it will miss event(s), but each subscriber handles this scenario and will go into catch-up mode. The publisher is there to notify the subscribers of new events to reduce latency. Rather than have them poll the database.

**Comment by slashdotdash** [comment]
This issue may be resolved by the work in progress to use a single writer process. As that will guarantee ordering of persisted events. Then there would be no need to have the publisher concern itself with that.

**Comment by stuartc** [comment]
Would it make sense to have a publisher+subscriber pair for each "subscription"?
Since the Subscriptions module handles `last_seen_event_id` quite nicely, and as much as there is quite a bit of overlap with Publisher (at least in my understanding) I really do like the separation.
Having a publisher for each subscriber could lead to a neat encapsulation of "where were we" and "what's next" I think. 
But, as you said the new single writer would make the order tracking obsolete.

In the short term I'm going to revert to my (awkwardly dangerous) gproc publisher. Would love to get your thoughts and ideas around the public API for Subscribers - hoping I can contribute in that space.

**Comment by slashdotdash** [comment]
Fixed by #9.
